src/action/macos/unmount_apfs_volume.rs

```diff
@@ -29,6 +28,26 @@ impl UnmountApfsVolume {
         let disk = disk.as_ref().to_owned();
         Ok(Self { disk, name }.into())
     }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn plan_skip_if_already_mounted_to_nix(
+        disk: impl AsRef<Path>,
+        name: String,
+    ) -> Result<StatefulAction<Self>, ActionError> {
+        let diskinfo = DiskUtilInfoOutput::for_volume_name(&name).await;
+
+        let mut task = Self::plan(&disk, name).await?;
+
+        if let Ok(diskinfo) = diskinfo {
+            if Path::new(&diskinfo.parent_whole_disk) == disk.as_ref()
+                && diskinfo.mount_point.as_deref() == Some("/nix".as_ref())
+            {
+                task.state = crate::action::ActionState::Skipped
```

Comment:
I don't really like that this is directly modifying the state rather than using `StatefulAction::skipped`, even though the result is the same, but I guess it's fine.